curriculum/challenges/english/07-scientific-computing-with-python/scientific-computing-with-python-projects/budget-app.md

```diff
@@ -8,22 +8,85 @@ dashedName: budget-app
 
-After going to that link, fork the project. Once you complete the project based on the instructions in 'README.md', submit your project link below.
+- [Python for Everybody Video Course](https://www.freecodecamp.org/news/python-for-everybody/) (14 hours)
+			
+- [Learn Python Video Course](https://www.freecodecamp.org/news/learn-python-video-course/) (10 hours)
 
-We are still developing the interactive instructional part of the Python curriculum. For now, here are some videos on the freeCodeCamp.org YouTube channel that will teach you everything you need to know to complete this project:
+# --instructions--
+
+Complete the `Category` class in `budget.py`. It should be able to instantiate objects based on different budget categories like *food*, *clothing*, and *entertainment*. When objects are created, they are passed in the name of the category. The class should have an instance variable called `ledger` that is a list. The class should also contain the following methods:
+
+- A `deposit` method that accepts an amount and description. If no description is given, it should default to an empty string. The method should append an object to the ledger list in the form of `{"amount": amount, "description": description}`.
+- A `withdraw` method that is similar to the `deposit` method, but the amount passed in should be stored in the ledger as a negative number. If there are not enough funds, nothing should be added to the ledger. This method should return `True` if the withdrawal took place, and `False` otherwise.
+- A `get_balance` method that returns the current balance of the budget category based on the deposits and withdrawals that have occurred.
+- A `transfer` method that accepts an amount and another budget category as arguments. The method should add a withdrawal with the amount and the description "Transfer to [Destination Budget Category]". The method should then add a deposit to the other budget category with the amount and the description "Transfer from [Source Budget Category]". If there are not enough funds, nothing should be added to either ledgers. This method should return `True` if the transfer took place, and `False` otherwise.
+- A `check_funds` method that accepts an amount as an argument. It returns `False` if the amount is greater than the balance of the budget category and returns `True` otherwise. This method should be used by both the `withdraw` method and `transfer` method.
+
+When the budget object is printed it should display:
+- A title line of 30 characters where the name of the category is centered in a line of `*` characters.
+- A list of the items in the ledger. Each line should show the description and amount. The first 23 characters of the description should be displayed, then the amount. The amount should be right aligned, contain two decimal places, and display a maximum of 7 characters.
+- A line displaying the category total.
+
+Here is an example of the output:
+
+```py
```

Comment:
@nhcarrigan Do we maybe want this to be `bash` or `console`?

Reply:
`bash` or `shell` could work. `console` isn't supported by prismjs.

Reply:
I did use `py` for the other project.

Reply:
`bash` and `shell` look the same. So, I think we can just go with bash, because it is used more in our codebase:
![image](https://user-images.githubusercontent.com/51722130/140413084-fd27aed5-c730-4cc8-8d5d-7c9334319609.png)

Reply:
```suggestion
```bash
```